plugins/plugin.go

```diff
@@ -199,6 +201,45 @@ func (p *Plugin) getContainerList() ([]string, error) {
 
 /*** Helpers ***/
 
+// mergeFutures aggregates multiple channels into a single output channel
+func mergeFutures(cs []<-chan producers.MetricsMessage) <-chan producers.MetricsMessage {
+	var wg sync.WaitGroup
+	out := make(chan producers.MetricsMessage)
+
+	// Start an output goroutine for each input channel in cs.  output
+	// copies values from c to out until c is closed, then calls wg.Done.
+	output := func(c <-chan producers.MetricsMessage) {
+		for n := range c {
+			out <- n
+		}
+		wg.Done()
+	}
+	wg.Add(len(cs))
+	for _, c := range cs {
+		go output(c)
+	}
+
+	// Start a goroutine to close out once all the output goroutines are
+	// done.  This must start after the wg.Add call.
+	go func() {
+		wg.Wait()
+		close(out)
+	}()
+	return out
+}
+
+// makeMetricsRequestFuture wraps makeMetricsRequest in a goroutine, returning
+// a channel on which the MetricsMessage will be returned.
+func makeMetricsRequestFuture(client *http.Client, url string) <-chan producers.MetricsMessage {
+	out := make(chan producers.MetricsMessage, 1)
+	go func() {
+		message, _ := makeMetricsRequest(client, url)
```

Comment:
Why are we ignoring the `err` return here?  Typically i would create a struct like

```
type messageAndErr struct {
  message producers.MetricsMessage
  err error
}

```

And then create a chan of these structs to return both the message and the error.  This way you do not have to ignore the error.